Short-Answer/Algorithms_Answers.md

```diff
@@ -2,13 +2,13 @@
 
 ## Exercise I
 
-a)
+a) O(n), this question only requires one loop
 
 
-b)
+b) O(n^2), this question has a nested loop as well. two loops total
```

Comment:
Answer: O(n log(n))

When thinking about "How many times will this loop run as n gets bigger?", we have to consider both loops individually. The outer for loop will run n times, since it is incrementing i by 1 each iteration.

However, the inner loop is doubling the value of j every time it runs. This means that the number of times the inner while loop runs is not growing at a linear rate - we would describe its runtime as O(log(n)).

Since these loops are nested, we multiply the individual runtimes of each loop to obtain the runtime of the entire block, giving us the overall runtime of O(n log(n)).